### PR TITLE
Validator improvements

### PR DESCRIPTION
**Pull request by PlasmaPower**
- [x] Fix inactive staker strategies
- [x] Withdraw funds from rollup contract
- [ ] Only create validator wallet when needed

### CODE REVIEW

File: packages/arb-node-core/staker/staker.go [hunk 1]
```diff
@@ -210,10 +223,27 @@ func (s *Staker) Act(ctx context.Context) (*arbtransaction.ArbTransaction, error
...
+	if s.bringActiveUntilNode != nil {
+		if info.LatestStakedNode.Cmp(s.bringActiveUntilNode) < 0 {
+			if effectiveStrategy == DefensiveStrategy {
+				effectiveStrategy = StakeLatestStrategy
+			}
+			s.inactiveLastCheckedNode = nil
+		} else {
+			logger.Info().Msg("defensive validator staked past incorrect node; waiting here")
+			s.bringActiveUntilNode = nil
+			s.inactiveLastCheckedNode = nil
```
Comment (hkalodner) [inline packages/arb-node-core/staker/staker.go:241]: Probably clearer to lift this out of the if statement since it's the same in both clauses

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-node-core/staker/staker.go [hunk 2]
```diff
@@ -268,6 +298,16 @@ func (s *Staker) Act(ctx context.Context) (*arbtransaction.ArbTransaction, error
...
+	withdrawable, err := s.rollup.WithdrawableFunds(ctx, s.wallet.Address())
+	if err != nil {
+		return nil, err
+	}
+	if withdrawable.Sign() > 0 && s.withdrawDestination != (common.Address{}) {
+		err = s.rollup.WithdrawFunds(ctx, s.withdrawDestination)
+		if err != nil {
+			return nil, err
+		}
+	}
```
[the remark this answers is not in the document]
Comment (PlasmaPower) [reply packages/arb-node-core/staker/staker.go:310]: I think we should think a bit more about how the authentication there would work. We'd either need to modify the rollup to only allow withdrawing to the validator itself, which seems a bit awkward, or have the validator wallet look at the function selector and filter based on that, which also seems a bit awkward.

**Review by hkalodner — DISMISSED** [review]
LGTM

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**